src/midictl.c

```diff
@@ -482,13 +482,50 @@ int main(int argc, char *argv[])
 		exit(EXIT_FAILURE);
 	}
 
+	bool save_config_path = 1;
+	// check if config file path is given
+	//if not open the last given path.
+	if (config.config_path == NULL)
+	{
+		FILE *prev_config_path = fopen(".prevconfpath", "rt");
+		if (prev_config_path == NULL)
+		{
+			//prev file doesnt exist
+			fprintf(stderr, "Previously used config path is non existant\n");
```

Comment:
I feel like this error message may be a little bit unclear when you run the program for the first time. Maybe something like `No config path specified and the previous config path file doesn't exists.` would be better? Printing the help message from `argp_usage()` here would be nice too, I think.

Reply:
I adjusted the error message to your suggestion. One thing though, I'm not familial with the argp helper functions. I couldn't quickly figure out how I should call argp_usage() after the printf statement. I'll research and edit when I'm able to figure it out.

Reply:
Oh, sorry, I forgot that `argp_usage()` takes a pointer to `argp_state`. That makes it a bit more tricky. You can try to find a workaround, but I think the new error message should be good enough,